### PR TITLE
chore(ci): test with tox

### LINKED ISSUE #329: Test with tox
Test with tox

### PR DESCRIPTION
**Pull request by justinchuby**
This PR introduces tox for managing test environments. It allows us to run the same tests locally and on CI on many different environment configurations.

- fixes #329
- Fix a new error in `cat` from torch-nightly

### CODE REVIEW

File: onnxscript/test/function_libs/torch_aten/ops_correctness_test.py [hunk 1]
```diff
@@ -148,9 +148,12 @@ def wrapped(fn):
 def duplicate_opinfo(opinfos: list[opinfo_core.OpInfo], name: str, new_names: tuple[str, ...]):
     """Duplicate an opinfo in the opinfo database and give it a new name."""
     duplicated = []
+    all_info_names = {opinfo.name for opinfo in opinfos}
     for opinfo in opinfos:
         if opinfo.name == name:
             for new_name in new_names:
+                if new_name in all_info_names:
+                    warnings.warn(f"OpInfo {new_name} already exists in the database.")
```
Comment (BowenBao) [inline onnxscript/test/function_libs/torch_aten/ops_correctness_test.py:156]: NOTE: This is solving an issue caused by changes in latest upstream pytorch
```
ERROR onnxscript/test/function_libs/torch_aten/ops_correctness_test.py - AssertionError: Redefinition of test test_output_match_full_cpu_float32
```

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -50,10 +50,6 @@ jobs:
           python -m pip install --upgrade pip
           python -m pip install --upgrade setuptools
           python -m pip install -q -r requirements-dev.txt
-          # Install ONNX
```
Comment (BowenBao) [inline .github/workflows/lint.yaml:53]: love this

File: azure-pipelines.yml [hunk 3]
```diff
@@ -10,45 +8,10 @@ pool:
...
-  displayName: 'pytest'
-
-- script: |
-    python setup.py install
-  displayName: 'install package'
-
-- script: |
-      if [ '$(test.examples)' == '1' ]
-        then
-          pytest -v docs/test
```
Comment (BowenBao) [inline azure-pipelines.yml:52]: I assume this will always be tested?

**Review by BowenBao — APPROVED** [review]
🚀 LGTM, please consult with @gramalingam regarding the autopep8 changes.

**Comment by justinchuby** [comment]
Looks like some tests were not run. @gramalingam did I miss a step for running the function tests?

**Comment by justinchuby** [comment]
I think it's default values which hints the onnx nightly packages where not used

**Comment by justinchuby** [comment]
that's it 

image

**Comment by justinchuby** [comment]
> 🚀 LGTM, please consult with @gramalingam regarding the autopep8 changes.

I reverted changes to autopep8

File: onnxscript/test/function_libs/torch_aten/ops_correctness_test.py (hunk 1 above)
Comment (justinchuby) [reply onnxscript/test/function_libs/torch_aten/ops_correctness_test.py:157]: Added a note

**Comment by justinchuby** [comment]
good
image

File: azure-pipelines.yml [hunk 4]
```diff
@@ -12,43 +10,8 @@ strategy:
   matrix:
     Python310-onnx:
       python.version: '3.10'
-      onnx.standard: '1'
-      test.examples: '1'
 steps:
 - task: UsePythonVersion@0
   inputs:
     versionSpec: '$(python.version)'
   displayName: 'Use Python $(python.version)'
```
Comment (jcwchen) [inline azure-pipelines.yml:17]: I might miss some context here: is it intended that after this PR this azure-pipelines seems to test nothing?

Comment (justinchuby) [reply azure-pipelines.yml:17]: Yes. We just needed it so ms can inject mandatory checks

**Pull request merged**